controllers/envoyfleet_resources.go

```diff
@@ -0,0 +1,320 @@
+	// Depends on the Service
+	f.generateService()
+
+	return f, nil
+}
+
+func (e *EnvoyFleetResources) CreateOrUpdate(ctx context.Context) error {
+	if err := k8sutils.CreateOrReplace(ctx, e.client, e.configMap); err != nil {
+		return fmt.Errorf("failed to deploy Envoy Fleet config map: %w", err)
+	}
+	if err := k8sutils.CreateOrReplace(ctx, e.client, e.deployment); err != nil {
+		return fmt.Errorf("failed to  deploy Envoy Fleet deployment: %w", err)
+	}
+	if err := k8sutils.CreateOrReplace(ctx, e.client, e.service); err != nil {
+		return fmt.Errorf("failed to deploy Envoy Fleet service: %w", err)
+	}
+	return nil
+}
+
+func (e *EnvoyFleetResources) generateConfigMap(ctx context.Context) error {
+	// future object labels
+	labels := map[string]string{
+		"app.kubernetes.io/component": "envoy-config",
+	}
+	// Copy over shared labels map
+	for key, value := range e.sharedLabels {
+		labels[key] = value
+	}
+
+	configMapName := "kusk-gateway-envoy-config-" + e.fleet.Name
+
+	xdsLabels := map[string]string{"app.kubernetes.io/name": "kusk-gateway", "app.kubernetes.io/component": "xds-service"}
+	xdsServices, err := k8sutils.GetServicesByLabels(ctx, e.client, xdsLabels)
+	if err != nil {
+		return fmt.Errorf("cannot create Envoy Fleet %s config map: %w", e.fleet.Name, err)
+	}
+	switch svcs := len(xdsServices); {
+	case svcs == 0:
+		return fmt.Errorf("cannot create Envoy Fleet %s config map: no xds services detected in the cluster when searching with the labels %s", e.fleet.Name, xdsLabels)
+	case svcs > 1:
+		return fmt.Errorf("cannot create Envoy Fleet %s config map: multiple xds services detected in the cluster when searching with the labels %s", e.fleet.Name, xdsLabels)
+	}
+	// At this point - we have exactly one service with (we ASSUME!) one port
+	xdsServiceHostname := fmt.Sprintf("%s.%s.svc.cluster.local.", xdsServices[0].Name, xdsServices[0].Namespace)
+	xdsServicePort := xdsServices[0].Spec.Ports[0].Port
+
+	e.configMap = &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            configMapName,
+			Namespace:       e.fleet.Namespace,
+			Labels:          labels,
+			OwnerReferences: []metav1.OwnerReference{envoyFleetAsOwner(e.fleet)},
+		},
+		Data: map[string]string{
+			"envoy-config.yaml": fmt.Sprintf(envoyConfigTemplate, e.fleet.Name, xdsServiceHostname, xdsServicePort),
+		},
+	}
+
+	return nil
+}
+
+func (e *EnvoyFleetResources) generateDeployment() {
+	// future object labels
+	labels := map[string]string{
+		"app.kubernetes.io/component": "envoy",
+	}
+	// Copy over shared labels map
+	for key, value := range e.sharedLabels {
+		labels[key] = value
+	}
+
+	deploymentName := "kusk-gateway-envoy-" + e.fleet.Name
+
+	configMapName := e.configMap.Name
+
+	// Create container template first
+	envoyContainer := corev1.Container{
+		Name:            "envoy",
+		Image:           e.fleet.Spec.Image,
+		ImagePullPolicy: "IfNotPresent",
```

Comment:
Is there not a constant we can use in place of this from one of the k8s modules?

Reply:
Thanks, fixed